src/Umbraco.Web/ImageCropperTemplateExtensions.cs

```diff
@@ -239,6 +239,95 @@ public static string GetCropUrl(this IPublishedContent mediaItem, string propert
+        /// </summary>
+        /// <param name="cropDataSet">
+        /// The ImageCropDataSet.
+        /// </param>
+        /// <param name="width">
+        /// The width of the output image.
+        /// </param>
+        /// <param name="height">
+        /// The height of the output image.
+        /// </param>
+        /// <param name="propertyAlias">
+        /// Property alias of the property containing the Json data.
+        /// </param>
+        /// <param name="cropAlias">
+        /// The crop alias.
+        /// </param>
+        /// <param name="quality">
+        /// Quality percentage of the output image.
+        /// </param>
+        /// <param name="imageCropMode">
+        /// The image crop mode.
+        /// </param>
+        /// <param name="imageCropAnchor">
+        /// The image crop anchor.
+        /// </param>
+        /// <param name="preferFocalPoint">
+        /// Use focal point, to generate an output image using the focal point instead of the predefined crop
+        /// </param>
+        /// <param name="useCropDimensions">
+        /// Use crop dimensions to have the output image sized according to the predefined crop sizes, this will override the width and height parameters.
+        /// </param>
+        /// <param name="cacheBuster">
+        /// Add a serialised date of the last edit of the item to ensure client cache refresh when updated
+        /// </param>
+        /// <param name="furtherOptions">
+        /// These are any query string parameters (formatted as query strings) that ImageProcessor supports. For example:
+        /// <example>
+        /// <![CDATA[
+        /// furtherOptions: "&bgcolor=fff"
+        /// ]]>
+        /// </example>
+        /// </param>
+        /// <param name="ratioMode">
+        /// Use a dimension as a ratio
+        /// </param>
+        /// <param name="upScale">
+        /// If the image should be upscaled to requested dimensions
+        /// </param>        
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public static string GetCropUrl(
+             this ImageCropDataSet cropDataSet,
+             int? width = null,
+             int? height = null,
+             string cropAlias = null,
+             int? quality = null,
+             ImageCropMode? imageCropMode = null,
+             ImageCropAnchor? imageCropAnchor = null,
+             bool preferFocalPoint = false,
+             bool useCropDimensions = false,
+             bool cacheBuster = true,
+             string cacheBusterValue = null,
+             string furtherOptions = null,
+             ImageCropRatioMode? ratioMode = null,
+             bool upScale = true)
+        {
+            if (cropDataSet.Udi == null) throw new ArgumentNullException("cropDataSet.Udi");
+
+            var mediaId = ApplicationContext.Current.Services.EntityService.GetIdForUdi(cropDataSet.Udi);
```

Comment:
I'm not sure using both the `ApplicationContext.Current` and `UmbracoContext.Current` singletons and the `EntityService` (although cached) are desirable here...